### PR TITLE
[T31] add imagePickerBottomSheet

### PR DESCRIPTION
**Pull request by idaosk**
**add addPicBottomSheet** 

- user should be able to click on the open camera button and take a picture, and by clicking ok should navigate back to add event bottom sheet and imageView should be set to the captured photo.

- user should be able to click on open gallery, pick a photo, click ok and navigate back to add event, and have the imageView set to the chosen photo from the gallery. 

### this is why I chose to use intents instead of cameraX:

1. I needed basic camera functionality 
2. cameraX provides support for advanced features like HDR, portrait mode, and night mode but we don't need them

### CODE REVIEW

**Comment by amirziarati** [comment]
i'm gonna ask you to review your own PR as a practice.
please review it as its someone else's
thanks

File: app/src/main/res/layout/bottom_sheet_add_event.xml [hunk 1]
```diff
@@ -10,8 +10,9 @@
 
         <ImageButton
             android:id="@+id/img_event"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
+            android:layout_width="120dp"
```
Comment (idaosk) [inline app/src/main/res/layout/bottom_sheet_add_event.xml:13]: no number literals. all numbers should come from dimen.xml

File: app/src/main/java/com/gathering/android/event/myevent/addevent/pic/AddPicBottomSheet.kt [hunk 2]
```diff
@@ -1,29 +1,151 @@
...
+    private var uriString: String? = null
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (requestCode == GALLERY_REQUEST_CODE && resultCode == Activity.RESULT_OK) {
+            data?.data?.let { uri ->
+                binding.imageView.setImageURI(uri)
+                uriString = uri.toString()
+            }
+        } else if (requestCode == CAMERA_REQUEST_CODE && resultCode == Activity.RESULT_OK) {
+            val imageBitmap = data?.extras?.get("data") as Bitmap
+
```
[the remark this answers is not in the document]
Comment (amirziarati) [reply app/src/main/java/com/gathering/android/event/myevent/addevent/pic/AddPicBottomSheet.kt:92]: use two buttons to let user rotate the image if needed.

File: app/src/main/java/com/gathering/android/event/myevent/addevent/pic/AddPicBottomSheet.kt [hunk 3]
```diff
@@ -1,29 +1,151 @@
...
+    private fun allPermissionsGranted(permList: Array<String>) = permList.all {
+        requireContext().checkSelfPermission(
+            it
+        ) == PackageManager.PERMISSION_GRANTED
+    }
+
+    private var uriString: String? = null
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (requestCode == GALLERY_REQUEST_CODE && resultCode == Activity.RESULT_OK) {
```
[the remark this answers is not in the document]
Comment (amirziarati) [reply app/src/main/java/com/gathering/android/event/myevent/addevent/pic/AddPicBottomSheet.kt:85]: with fast CPUs today is more like a code-style choice but technically the more cases get the better it is to use a `when` statement (or `switch` statement in java) over a if-else.

in a 10 cases `if-else` it takes cpu 10 actions for the worst scenario to find the right block to run. (**complexity of O(n)**)
using `when` or `switch-case` for the same 10 cases, it only takes 1 action  (**complexity of O(1)**)

File: app/src/main/java/com/gathering/android/event/myevent/addevent/AddEventBottomSheetFragment.kt [hunk 4]
```diff
@@ -31,9 +33,16 @@ class AddEventBottomSheetFragment : BottomSheetDialogFragment() {
...
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
-    ): View? {
+    ): View {
         binding = BottomSheetAddEventBinding.inflate(LayoutInflater.from(requireContext()))
 
+        val selectedImagePath = arguments?.getString(KEY_ARGUMENT_SELECTED_IMAGE)
+
+
+        if(selectedImagePath != null) {
```
Comment (moyarsuccess) [inline app/src/main/java/com/gathering/android/event/myevent/addevent/AddEventBottomSheetFragment.kt:42]: Not a big deal, but
This file needs to be reformatted, look at the `if(` it needs a space that shows this file is not formatted.

File: app/src/main/java/com/gathering/android/event/myevent/addevent/pic/AddPicBottomSheet.kt [hunk 5]
```diff
@@ -1,29 +1,151 @@
...
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int, permissions: Array<String>, grantResults: IntArray
+    ) {
+        if (requestCode == CAMERA_PERMISSION_CODE) {
+            if (allPermissionsGranted(REQUIRED_CAMERA_PERMISSIONS)) {
+                openCamera()
+            } else {
+                Toast.makeText(requireContext(), "camera permission denied", Toast.LENGTH_LONG)
```
[the remark this answers is not in the document]
Comment (moyarsuccess) [reply app/src/main/java/com/gathering/android/event/myevent/addevent/pic/AddPicBottomSheet.kt:136]: For permission granting, @Animansoubi has added a third-party lib which makes this so much easier, for now, you can merge it not to have a conflict but later you can migrate this to that lib too.

**Pull request merged**